tests/esm-child-process/process-forking-nested-esm-node-next/index.mts

```diff
@@ -0,0 +1,23 @@
+import { fork } from 'child_process';
+import { dirname, join } from 'path';
+import { fileURLToPath } from 'url';
+
+// Initially set the exit code to non-zero. We only set it to `0` when the
+// worker process finishes properly with the expected stdout message.
+process.exitCode = 1;
+
+const projectDir = dirname(fileURLToPath(import.meta.url));
+const workerProcess = fork(join(projectDir, 'worker.mts'), [], {
```

Comment:
Why does this test case use an absolute path for the worker's entrypoint?  Is it testing something that hasn't occurred to me?  I figure it should be same as the other tests.

Reply:
See response above. One test, with `.mts` I used `import.meta.url` to construct an absolute path to the worker. The other tests were not using `import.meta` (and couldn't --without transpile-only) and use a relative path.

Reply:
Ok cool, sounds like all the things we need to test are:

1. entrypoint script runs with correct `process.cwd()`
2. `fork()` entrypoint resolves with absolute path
3. `fork()` entrypoint resolves with relative path relative to `process.cwd()`, not parent's `ts-node --cwd` flag

...for both CJS and ESM, with 3. disabled in the non-ESM case since that bug will be fixed by other tickets.

I'll update the PR now.